### PR TITLE
Use CRO to create s3 bucket for 3scale

### PR DESCRIPTION
**Pull request by dimitraz**
https://issues.jboss.org/browse/INTLY-3092

### CODE REVIEW

File: pkg/controller/installation/products/threescale/reconciler.go [hunk 1]
```diff
@@ -332,6 +299,146 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient pkgcl
...
+			credSec.Data["AWS_ACCESS_KEY_ID"] = blobStorageSec.Data["credentialKeyID"]
+			credSec.Data["AWS_SECRET_ACCESS_KEY"] = blobStorageSec.Data["credentialSecretKey"]
+
+			err = serverClient.Create(ctx, credSec)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create blob storage aws credentials secret")
+			}
+		}
+
+		// return the file storage spec
```
Comment (aidenkeating) [inline pkg/controller/installation/products/threescale/reconciler.go]: From here on, we could do

- read operator sec
- createorupdate namespace sec using data from operator sec

that will flatten this entire section

i'd also split these into two funcs and move the external resource conditional up a level

**Comment by aidenkeating** [comment]
this looks great, going to give it a spin once I have access to my cluster again?

i know you're away tomorrow, are you good with this being merged if verification goes ok while you're away?

**Comment by dimitraz** [comment]
@aidenkeating Sure if you can that would be great!

**Pull request merged**